x-pack/examples/ui_actions_enhanced_examples/public/dashboard_hello_world_drilldown/index.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiFormRow, EuiFieldText } from '@elastic/eui';
+import { reactToUiComponent } from '../../../../../src/plugins/kibana_react/public';
+import { UiActionsEnhancedDrilldownDefinition as Drilldown } from '../../../../plugins/advanced_ui_actions/public';
+import {
+  EmbeddableContext,
+  RangeSelectTriggerContext,
+  ValueClickTriggerContext,
+} from '../../../../../src/plugins/embeddable/public';
+import { CollectConfigProps } from '../../../../../src/plugins/kibana_utils/public';
+
+export type ActionContext = RangeSelectTriggerContext | ValueClickTriggerContext;
```

Comment:
> I think this maybe should be just an empty object because this drilldown doesn't care about any context shape?

This is a simple "Hello world" example drilldown, which indeed only shows an alert and does not do much. But if this drilldown did something with the context, those would be the correct types to use.

I think it is useful to have those types there as when somebody would want to build a drilldown based on that sample drilldown they will know the right types.

> The typing here is probably not very safe [...] you could have a drilldown that says "I need X action context" but not actually get it ...

I'm not sure it qualifies as "not very safe" and that is just how JavaScript works.

Same goes for pretty much everything we do. You can define your NP plugin with any types you want, but it does not mean you will actually receive them. It is up to you to make sure that the data you receive is actually of the type you specified.

```ts
class MyPlugin implements Plugin<A, B, C, D> {

}
```

Or you can define an embeddable with any input you want, but it does not mean it will actually receive it.

```ts
class MyEmbeddable extends Embeddable<A> {

}
```

etc.